Donation-platform/src/main/java/com/google/model/Users.java

```diff
@@ -0,0 +1,77 @@
+package com.google.model;
+
+import java.util.List;
+import java.util.Objects;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+
+
+/** Represents a Users : id, userName, email, userInterests, charitiesDonatedTo. */
+public final class Users {
+  
+  // Datastore key uniquely identifying user.
+  private Key id;
+  // Username of user
+  private String userName;
+  // Email address of user 
+  private String email;
+  // Tag IDs representing category tags the user selected that they were interested in.
+  private List<Key> userInterests;
+  // Charity IDs representing charities the user donated to.
+  private List<Key>  charitiesDonatedTo;
+
+  // Initialize all fields of Users
+  public Users(Key id, String userName, String email,
+              List<Key> userInterests, List<Key>  charitiesDonatedTo) {
+    this.id = id;
+    this.userName = userName;
+    this.email = email;
+    this.userInterests = userInterests;
+    this.charitiesDonatedTo = charitiesDonatedTo;
+  }
+  // [END fs_class_definition]
+
+  public Users(Key id) {
+    this.id = id;
+  }
+
+  public Key getId() {
```

Comment:
this does not take any param, on what basis it will return id of a user?

Reply:
the id for the user is set in the database. That is what is returned.